### ISSUE TITLE
(Elementary) shortest path problem with resource constraints

### ISSUE THREAD

**Issue by ruthmair** [Gurobi/gurobi-optimods#50]
**Why this Mod?**

- Shortest path problem from given source node to target node minimizing total arc and/or node costs, subject to knapsack constraints for the whole path on further resources like distance/time/load on arcs and/or nodes.
- Costs are allowed to be negative potentially leading to loops along the path (even infinite loops if resource constraints are not restricting). If asking for "elementary" paths, loops are forbidden.
- Classical NP-hard combinatorial problem, often arising as subproblem of more complicated problems, e.g., in the pricing problem of route-based formulations for vehicle routing problems.
- State-of-the-art algorithms are labeling algorithms similar to dynamic programming that can be difficult to implement correctly and efficiently. For inefficient implementations, MILP might have comparable performance.

**What will the API be?**

- Graph-based input seems more appropriate instead of multiple matrices/arrays that may be inconsistent. Pandas dataframes are a good option.
- Resource "cost" is mandatory and serves as resource to minimize. All further resources with arbitrary names are optional for nodes and arcs.
- Node and arc values with the same resource name are accumulated along the path.

```
node_data = pd.DataFrame([
    {"node": 0, "cost": -3, "time": 4, "distance": 3},
    {"node": 1, "cost": 5, "time": 3, "distance": 7},
    {"node": 2, "cost": 6, "time": 1, "distance": 2},
    {"node": 3, "cost": -4, "time": 5, "distance": 3}
])

arc_data = pd.DataFrame([
    {"source": 0, "target": 1, "cost": 16, "distance": 1},
    {"source": 0, "target": 2, "cost": -13, "distance": 3},
    {"source": 1, "target": 2, "cost": 10, "distance": 2},
    {"source": 2, "target": 1, "cost": -4, "distance": 6},
    {"source": 1, "target": 3, "cost": 12, "distance": 3},
    {"source": 3, "target": 2, "cost": -9, "distance": 2}
])

limits = {
  "time": 10,
  "distance": 18
}

path, objective = solve_espprc(node_data=node_data, arc_data=arc_data, source=0, target=3, limits=limits, elementary=True)
```

**Additional context**

**Comment by ruthmair** [comment]
I guess the public/private issue also made the rcsp fork unusable? This is no issue at all, I can just start with a new fork.

**Comment by simonbowly** [comment]
Yeah, new fork required I'm afraid, sorry about that.

**Comment by torressa** [comment]
@ruthmair Are you still working on this? Simon has given us the OK on writing a VRP mod! 😈

**Comment by ruthmair** [comment]
I still need to start working on this, but it is on my list.
Would you do a VRP mod based on column generation?

**Comment by torressa** [comment]
Yeah, why not? Would you say the formulation is better? We could have mod based on the formulation(s), or even better formulation(s) + heuristic to insert solutions via callback
